configure.py

```diff
@@ -1887,7 +1887,23 @@ def write_build_file(f,
         rule wasm2wat
             command = wasm2wat $in > $out
             description = WASM2WAT $out
-        ''').format(**globals()))
+        ''').format(configure_args=configure_args,
+                    outdir=outdir,
+                    cxx=cxx,
+                    user_cflags=user_cflags,
+                    distro_extra_cflags=distro_extra_cflags,
+                    warnings=warnings,
+                    defines=defines,
+                    linker_flags=linker_flags,
+                    user_ldflags=user_ldflags,
+                    distro_extra_ldflags=distro_extra_ldflags,
+                    libs=libs,
+                    link_pool_depth=link_pool_depth,
+                    python=python,
+                    seastar_path=args.seastar_path,
+                    ninja=ninja,
+                    ragel_exec=ragel_exec))
```

Comment:
Better yet, we could use `f"{variable_name}"` and rid ourselves of this boilerplate.

Reply:
agreed, i'd go this way if we would want to keep this piece of code longer. but my goal here is to make the data dependency more obvious so i can either remove more global variables or extract them into a function, and eventually. and the whole process can be audited by reviewers without lots of pain. we will have a `configure.py` looks like:
```py
if __name__ == '__main__':
  args = parse_args()
  generate_build_file(args)
  generate_compdb(args)
```

so that i will be able to change it to:
```py
if __name__ == '__main__':
  args = parse_args()
  if args.use_cmake:
    generate_build_file_with_cmake(args)
  else:
    generate_build_file_with_python(args)
  generate_compdb(args)
```
reviewer would be able to tell if anything is missing in the new implementation by checking the code. currently the code is difficult to comprehend without reading through it couple times and making some notes.

and then remove a large chunk of the code for generating `build.ninja`   :sunglasses: :
```py
if __name__ == '__main__':
  args = parse_args()
  generate_build_file_with_cmake(args)
  generate_compdb(args)
```

Reply:
ok, I will queue as-is